Add a shared code sync action (#3)
Azure/dotnet-extensions-experimental: src/Shared/DotNetSync/ReadMe.SharedCode.md

> The code in this directory is shared between dotnet/extensions and azure/dotnet-extensions-experimental. This contains shared sources used in both repos. Any changes to this dir need to be checked into both repositories.

Reviewer: ```suggestion
The code in this directory is shared between [dotnet/extensions](https://github.com/dotnet/extensions) and [azure/dotnet-extensions-experimental](https://github.com/Azure/dotnet-extensions-experimental). This contains shared sources used in both repos. Any changes to this directory need to be checked into both repositories.
```

dotnet/extensions code paths:

> - extensions\src\shared\

Reviewer: ```suggestion
- [extensions\src\Shared\](https://github.com/dotnet/extensions/tree/main/src/Shared)
```

azure/dotnet-extensions-experimental code paths:

> - dotnet-extensions-experimental\src\shared\

Reviewer: ```suggestion
- [dotnet-extensions-experimental\src\Shared\](https://github.com/Azure/dotnet-extensions-experimental/tree/main/src/Shared)
```

## Copying code

> - To copy code from dotnet/extensions to azure/dotnet-extensions-experimental, set AZUREEXTENSIONS_REPO to the azure repo root and then run CopyToAzure.cmd.
> - To copy code from azure/dotnet-extensions-experimental to dotnet/extensions, set DOTNETEXTENSIONS_REPO to the dotnet repo root and then run CopyToDotnet.cmd.

Reviewer: ```suggestion
- To copy code from dotnet/extensions to azure/dotnet-extensions-experimental, set `AZUREEXTENSIONS_REPO` environment variable to the azure repo root and then run `CopyToAzure.cmd`.
- To copy code from azure/dotnet-extensions-experimental to dotnet/extensions, set `DOTNETEXTENSIONS_REPO` environment variable to the dotnet repo root and then run `CopyToDotnet.cmd`.
```

## GitHub Actions

> In azure/dotnet-extensions-experimental, the [dotnet-sync](https://github.com/Azure/dotnet-extensions-experimental/actions/workflows/dotnet-sync.yml) GitHub action automatically creates PRs to pull in changes from dotnet/runtime.

Reviewer: ```suggestion
In azure/dotnet-extensions-experimental, the [dotnet-sync](https://github.com/Azure/dotnet-extensions-experimental/actions/workflows/dotnet-sync.yml) GitHub action automatically creates PRs to pull in changes from dotnet/extensions.
```
